compiler/src/dotty/tools/dotc/typer/Inferencing.scala

```diff
@@ -81,14 +84,25 @@ object Inferencing {
       case tvar: TypeVar
       if !tvar.isInstantiated && ctx.typerState.constraint.contains(tvar) =>
         force.appliesTo(tvar) && {
+          if (force.implicitMode) {
+            // instantiate to or-dominator of lower bound; without this tweak we'd
+            // fail to find an implicit in situations like this:
+            //
+            //   def f[T: Ordering](xs: T*) = ...
+            //   f(Some(1), None)
+            //
+            // The problem is that there is no implicit Ordering instance for the otherwise inferred
+            // lower bound of T, which is `Some[Int] | None`.
+            ctx.orDominator(ctx.typeComparer.bounds(tvar.origin).lo) <:< tvar.origin
```

Comment:
Using a subtype check here to change how instantiation proceeds feels a bit too indirect. How about instead we add an `implicitMode: Boolean` parameter (or it could be called `widenUnion` maybe) to `TypeVar#instantiate` instead?